cli/cmd/install/install.go

```diff
@@ -0,0 +1,500 @@
+	}
+
+	// Setup logger to stream Helm library logs
+	var uiLogger = func(s string, args ...interface{}) {
+		logMsg := fmt.Sprintf(s, args...)
+		c.UI.Output(logMsg, terminal.WithInfoStyle())
+	}
+
+	// Set up the kubernetes client to use for non Helm SDK calls to the Kubernetes API
+	// The Helm SDK will use settings.RESTClientGetter for its calls as well, so this will
+	// use a consistent method to target the right cluster for both Helm SDK and non Helm SDK calls.
+	if c.kubernetes == nil {
+		restConfig, err := settings.RESTClientGetter().ToRESTConfig()
+		if err != nil {
+			c.UI.Output("Retrieving Kubernetes auth: %v", err, terminal.WithErrorStyle())
+			return 1
+		}
+		c.kubernetes, err = kubernetes.NewForConfig(restConfig)
+		if err != nil {
+			c.UI.Output("Initializing Kubernetes client: %v", err, terminal.WithErrorStyle())
+			return 1
+		}
+	}
+
+	c.UI.Output("Pre-Install Checks", terminal.WithHeaderStyle())
+
+	if err := c.checkForPreviousInstallations(settings, uiLogger); err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Ensure there's no previous PVCs lying around.
+	if err := c.checkForPreviousPVCs(); err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Ensure there's no previous bootstrap secret lying around.
+	if err := c.checkForPreviousSecrets(); err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Handle preset, value files, and set values logic.
+	vals, err := c.mergeValuesFlagsWithPrecedence(settings)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+	valuesYaml, err := yaml.Marshal(vals)
+	if err != nil {
+		c.UI.Output(err.Error(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// Print out the installation summary.
+	if !c.flagAutoApprove {
+		c.UI.Output("Consul Installation Summary", terminal.WithHeaderStyle())
+		c.UI.Output("Installation name: %s", defaultReleaseName, terminal.WithInfoStyle())
+		c.UI.Output("Namespace: %s", c.flagNamespace, terminal.WithInfoStyle())
+
+		if len(vals) == 0 {
+			c.UI.Output("Overrides: "+string(valuesYaml), terminal.WithInfoStyle())
+		} else {
+			c.UI.Output("Overrides:"+"\n"+string(valuesYaml), terminal.WithInfoStyle())
+		}
+	}
+
+	// Without informing the user, default global.name to consul if it hasn't been set already. We don't allow setting
+	// the release name, and since that is hardcoded to "consul", setting global.name to "consul" makes it so resources
+	// aren't double prefixed with "consul-consul-...".
+	vals = mergeMaps(convert(globalNameConsul), vals)
+
+	// Dry Run should exit here, no need to actual locate/download the charts.
+	if c.flagDryRun {
+		c.UI.Output("Dry run complete - installation can proceed.", terminal.WithInfoStyle())
+		return 0
+	}
+
+	if !c.flagAutoApprove {
+		confirmation, err := c.UI.Input(&terminal.Input{
+			Prompt: "Proceed with installation? (y/n)",
```

Comment:
I would capitalize either "y" or "n" depending on which choice will be run by default.

Reply:
good idea!